github/Authorization.py

```diff
@@ -136,45 +127,43 @@ def edit(
         :param note_url: string
         :rtype: None
         """
-        assert scopes is github.GithubObject.NotSet or all(
+        assert isinstance(scopes, _NotSetType) or all(
```

Comment:
I really don't like how this degenerates from

    assert scopes is NotSet

to

    assert isinstance(scopes, _NotSetType)

@JLLeitschuh any suggestions?

Reply:
https://github.com/python/typing/issues/236

Reply:
Oh yes, it impact performance.

But I don't think it's very necessary

```py
import timeit


class T:
    ...


class UnSetType(T):
    ...


unset = UnSetType()

print(timeit.timeit(lambda: unset is unset)) # 0.03375210000012885
print(timeit.timeit(lambda: isinstance(unset, UnSetType))) # 0.03873609999936889
```

and also `UnSet` is `Attribute[None]` now.

Reply:
using enum is possible but it will touch almost all files

Reply:
For now, let's go with what we have, and then would you be willing to open a follow-up pull request to attempt the enum changeover?

Reply:
I mean this aesthetically.

    assert scopes is NotSet

Is readable,

    assert isinstance(scopes, _NotSetType)

is ugly.

Plus, we need to import a private class `_NotSetType`. A lot of code has to be touched. The former was present before.

Reply:
But python typing system doesn't support this.

If I use emun as single value here, this can't pass mypy:

```py
o: Attribute[Optional[int]] = NotSet
```

Reply:
Also `Protocol` doesn't support generic, so make `Attribute` a `Protocol` won't work.

I don't know if there is any way to avoid all these problem.

Reply:
There is a good side of this, now you can write `assert isinstance(v, (_NotSetType, str))` instead of `assert v is NotSet or instance(v, str)`

Reply:
I agree with @trim21 here. There may not be a way to make the type checker happy moving forward. Im inclined to let us poke at trying to make this better in a followup PR